lib/AST/Builtins.cpp

```diff
@@ -281,6 +281,24 @@ static ValueDecl *getBinaryPredicate(Identifier Id, Type ArgType) {
   return getBuiltinFunction(Id, ArgElts, ResultTy);
 }
 
+static unsigned getLeastWidthOrPointerWidth(ASTContext &Context,
+                                            BuiltinIntegerType *BIT) {
+  if (BIT->getWidth().isPointerWidth()) {
+    return Context.LangOpts.getPointerWidth();
+  } else {
+    return BIT->getLeastWidth();
+  }
+}
+
+static unsigned getGreatestWidthOrPointerWidth(ASTContext &Context,
+                                               BuiltinIntegerType *BIT) {
+  if (BIT->getWidth().isPointerWidth()) {
+    return Context.LangOpts.getPointerWidth();
+  } else {
+    return BIT->getGreatestWidth();
+  }
+}
```

Comment:
I think you should also drop these changes.

The theoretical goal behind `Builtin.Word` is that pure Swift code which avoids target conditionals and uses portable types like it will produce Swift code — at the AST and SIL levels at least — that's actually target-independent.  We've gradually moved away from that being a useful goal of the primary stdlib implementation, but again, if we want to commit to that course we should completely drop `Builtin.Word`.  In the meantime, there shouldn't be anything short of IRGen that assumes an actual width for `Builtin.Word`.

Reply:
I've pulled that change and the associated unit tests. How does this much smaller PR look?

Reply:
p.s. Can someone also kick off a smoke test on this for osx and linux please?